tlspuffin/src/tls/fn_utils.rs

```diff
@@ -79,6 +80,115 @@ pub fn fn_decrypt_handshake(
+        .tls13()
+        .ok_or_else(|| FnError::Crypto("No tls 1.3 suite".to_owned()))?
+        .derive_decrypter(&key);
+    let message = decrypter
+        .decrypt(
+            PlainMessage::from(application_data.clone()).into_unencrypted_opaque(),
+            *sequence,
+        )
+        .map_err(|_err| FnError::Crypto("Failed to decrypt it fn_decrypt_handshake".to_string()))?;
+
+    let payloads =
+        MessagePayload::multiple_new(message.typ, message.version, message.payload).unwrap();
+
+    let messages = payloads
+        .into_iter()
+        .map(|p| Message {
+            version: message.version,
+            payload: p,
+        })
+        .collect();
+
+    Ok(messages)
+}
+
+pub fn fn_find_server_certificate(messages: &Vec<Message>) -> Result<Message, FnError> {
+    for msg in messages {
+        if let MessagePayload::Handshake(x) = &msg.payload {
+            if x.typ == HandshakeType::Certificate {
+                return Ok(msg.clone());
+            }
+        }
+    }
+    Err(FnError::Unknown("no server certificate".to_owned()))
+}
+
+pub fn fn_find_server_ticket(messages: &Vec<Message>) -> Result<Message, FnError> {
+    for msg in messages {
+        if let MessagePayload::Handshake(x) = &msg.payload {
+            if x.typ == HandshakeType::NewSessionTicket {
+                return Ok(msg.clone());
+            }
+        }
+    }
+    Err(FnError::Unknown("no server tickets".to_owned()))
+}
+
+pub fn fn_find_server_certificate_request(messages: &Vec<Message>) -> Result<Message, FnError> {
+    for msg in messages {
+        if let MessagePayload::Handshake(x) = &msg.payload {
+            if x.typ == HandshakeType::CertificateRequest {
+                return Ok(msg.clone());
+            }
+        }
+    }
+    Err(FnError::Unknown("no server tickets".to_owned()))
+}
+
+pub fn fn_find_encrypted_extensions(messages: &Vec<Message>) -> Result<Message, FnError> {
+    for msg in messages {
+        if let MessagePayload::Handshake(x) = &msg.payload {
+            if x.typ == HandshakeType::EncryptedExtensions {
+                return Ok(msg.clone());
+            }
+        }
+    }
+    Err(FnError::Unknown("no encrypted extensions".to_owned()))
+}
+
+pub fn fn_find_server_certificate_verify(messages: &Vec<Message>) -> Result<Message, FnError> {
+    for msg in messages {
+        if let MessagePayload::Handshake(x) = &msg.payload {
+            if x.typ == HandshakeType::CertificateVerify {
+                return Ok(msg.clone());
+            }
+        }
+    }
+    Err(FnError::Unknown("no certificate verify".to_owned()))
+}
+
+pub fn fn_find_server_finished(messages: &Vec<Message>) -> Result<Message, FnError> {
+    for msg in messages {
+        if let MessagePayload::Handshake(x) = &msg.payload {
+            if x.typ == HandshakeType::Finished {
+                return Ok(msg.clone());
+            }
+        }
+    }
+    Err(FnError::Unknown("no finished".to_owned()))
```

Comment:
More major discussion: Does it make sense to add one function symbol instead that extracts a message by type (e.g. `HandshakeType::Finished`) or index?

Reply:
We are thinking about this for a next PR (probably along the rework of the queries). For now, we are going to keep things like this for this PR